src/dotnet/APIView/ClientSPA/src/app/_components/api-revision-options/api-revision-options.component.ts

```diff
@@ -191,74 +194,89 @@ export class ApiRevisionOptionsComponent implements OnChanges {
     });
   }
 
-  identifyAndProcessSpecialAPIRevisions(mappedApiRevisions: any []) {
-    const result = [];
-    const semVarRegex = /(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:(?<presep>-?)(?<prelabel>[a-zA-Z]+)(?:(?<prenumsep>\.?)(?<prenumber>[0-9]{1,8})(?:(?<buildnumsep>\.?)(?<buildnumber>\d{1,3}))?)?)?/;
-
-    let latestGAApiRevision : any = null;
-    let latestApprovedApiRevision : any = null;
-    let currentMainApiRevision : any = null;
-    let latestReleasedApiRevision : any = null;
-
-    while (mappedApiRevisions.length > 0) {
-      let apiRevision = mappedApiRevisions.shift();
+  tagSpecialRevisions(mappedApiRevisions: any []) {
+    this.tagLatestGARevision(mappedApiRevisions);
+    this.tagLatestApprovedRevision(mappedApiRevisions);
+    this.tagCurrentMainRevision(mappedApiRevisions);
+    this.tagLatestReleasedRevision(mappedApiRevisions);   
+  }
 
-      if (latestGAApiRevision === null) {
-        if (apiRevision.version) {
-          let versionParts = apiRevision.version.match(semVarRegex);
+  tagLatestGARevision(apiRevisions: any[]) {
+    const gaRevisions : any [] = [];
 
-          if (versionParts.groups?.prelabel === undefined && versionParts.groups?.prenumber === undefined &&
-            versionParts.groups?.prenumsep === undefined && versionParts.groups?.presep === undefined) {
-              apiRevision.isLatestGA = true;
-              latestGAApiRevision = apiRevision;
-              continue;
-          }
+    for (let apiRevision of apiRevisions) {
+      if (apiRevision.version) {
+        const semVar = new AzureEngSemanticVersion(apiRevision.version, apiRevision.language);
+        if (semVar.versionType == "GA") {
+          apiRevision.semanticVersion = semVar;
+          gaRevisions.push(apiRevision);
         }
       }
+    }
 
-      if (latestApprovedApiRevision === null) {
-        if (apiRevision.isApproved) {
-          apiRevision.isLatestApproved = true;
-          latestApprovedApiRevision = apiRevision;
-          continue;
-        }
-      }
+    if (gaRevisions.length > 0) {
+      gaRevisions.sort((a: any, b: any) => a.semanticVersion.compareTo(b.semanticVersion));
+      gaRevisions[0].isLatestGA = true;
```

Comment:
We need to check if this revision is released. `Latest GA` means most recent released GA version. IF we don't check for released status then a GA unreleased version will be shown as latest GA.

Reply:
The way I read `Latest GA` is that it is the last GA sdk that was released. In some cases `Latest GA == Latest released` but this won't be true in all scenarios. The way we know an SDK is GA is because the version won't have a `-beta/b/etc` on it.
Is this something we already have here?

Reply:
I have updated the logic.